### PR TITLE
fix #24 #28 Windows right-click menu disappears

### LINKED ISSUE #24: [Bug?]`Clear Record` disappears quickly
Like the video below:\

When I right click on a cell to clear record, it disappeared quickly

https://user-images.githubusercontent.com/13215013/188766453-af1f7af9-7f02-4215-a0c0-231c52b0622e.mp4

### PR DESCRIPTION
**Pull request by GitMurf**
I still want to do some more digging maybe this weekend to figure out more of the "why" of the problem, but bottom-line the issue was that the right clicks were being done by a `mousedown` event which showed the menu but then on the mouse up (finishing of the click) Obsidian natively seems to then trigger the "contextmenu" event which then clears any menus that are already opened. I think there are some "timing issues" with it and also I think there are way too many event listeners running / conflicting overall that I want to look into another day but I at least have a solution 🙂 I would also like to see what Licat / other devs have to say about this potential issue with "custom context menus" because I seem to remember a warning a long time ago that it can get a little messy.

Basically I now turn the `mousedown` event into a trigger to add a context menu window event listener at which it adds the menu at that point on the context menu and then stops the default and propagation. So now the menu is actually not loading / firing officially until the `contextmenu` event which is after the mouse up and click events.

### CODE REVIEW

**Comment by marcusolsson** [comment]
Thank you for your hard work on this! Definitely a tricky one, but I'm happy with this solution for 1.0.

**Comment by GitMurf** [comment]
@marcusolsson btw I take no offense to re-structuring or renaming etc the variables/function names. I know how it is when others contribute and doesn’t quite line up with how you would do things. Just wanted to make that clear in case you prefer to implement my fix in another way :)

**Comment by marcusolsson** [comment]
I have no comments or ideas on how to improve it at the moment. I think it's clean and easy to understand, and don't mind merging as is 🙏  Only thing is I can't verify it on my machine.

I noticed that it's only for the menus in the Table view, I assume the other menus work as intended?
